shell/common/benchmark_D2F-native-spark.sh

```diff
@@ -0,0 +1,123 @@
+  NATIVE_DB="tpch_orc_${SCALE_FACTOR}"
+  logger "INFO: Setting INPUT_DIR to read from HIVE table"
+else
+  logger "WARN: NO INPUT_DIR SET"
+fi
+
+# Set output base dir; a subdirectory for each BENCH_CURRENT_NUM_RUN is created
+# Info: do not miss last "/"
+[ ! "$NATIVE_OUTPUT_DIR" ] && NATIVE_OUTPUT_DIR="/native_spark/output/"
+# Set the output format that; possible values json, csv, orc, parquet; default: json
+[ ! "$NATIVE_OUT_FORMAT" ] && NATIVE_OUT_FORMAT="json"
+# Set if print out; possible values 0 (off) or 1 (on); default: 1 - if "on" no writing to file
+[ ! "$NATIVE_SPRINT" ] && NATIVE_SPRINT="1"
+
+
+benchmark_suite_run() {
+  logger "INFO: Running $BENCH_SUITE"
+
+  tpc-h_datagen
+
+  BENCH_CURRENT_NUM_RUN="1" #reset the global counter
+
+  # Iterate at least one time
+  while true; do
+    [ "$BENCH_NUM_RUNS" ] && logger "INFO: Starting iteration $BENCH_CURRENT_NUM_RUN of $BENCH_NUM_RUNS"
+
+    for query in $BENCH_LIST ; do
+      logger "INFO: RUNNING Query $query -- current run: $BENCH_CURRENT_NUM_RUN"
+      execute_tpchquery_spark "$query"
+    done
+
+    # Check if requested to iterate multiple times
+    if [ ! "$BENCH_NUM_RUNS" ] || [[ "$BENCH_CURRENT_NUM_RUN" -ge "$BENCH_NUM_RUNS" ]] ; then
+      break
+    else
+      BENCH_CURRENT_NUM_RUN="$((BENCH_CURRENT_NUM_RUN + 1))"
+    fi
+  done
+
+  logger "INFO: DONE executing $BENCH_SUITE"
+}
+
+# $1 query number
+# jar is expecting 8 args [BenchNum, query, inputdir, outputdir, intputformat, outputformat, screenprint, filter]
+# BenchNum for data output_dir
+# query for TPCH query
+# inputdir for the data input
+# outputdir for data output
+# TODO add csv / jdbc
+# informat of inputdata [text, orc, json, parquet]
+# outformat of putputdata [text, orc, json, parquet]
+# TODO ADD filter
+# filter to turon filter pushdown on or off
+# Hive/tbl version expects 7 args [BenchNum, query, outputdir, outputformat, screenprint, filter, DB]
+execute_tpchquery_spark() {
+  local query="$1"
+
+  if [[ "$NATIVE_FORMAT" == "text" ]]; then
+    execute_spark "tpch_query_$query" "--class main.scala.TpchQuery $NATIVE_SPARK_LOCAL_DIR/spark-tpc-h-queries_2.11-1.0-txt.jar $BENCH_CURRENT_NUM_RUN $query $NATIVE_INPUT_DIR $NATIVE_OUTPUT_DIR $NATIVE_FORMAT $NATIVE_OUT_FORMAT $NATIVE_SPRINT $NATIVE_FILTER" "time"
+  elif [[ "$NATIVE_FORMAT" == "orc" || "$NATIVE_FORMAT" == "parquet" || "$NATIVE_FORMAT" == "json" ]]; then
+    execute_spark "tpch_query_$query" "--class main.scala.TpchQuery $NATIVE_SPARK_LOCAL_DIR/spark-tpc-h-queries_2.11-1.0-orc.jar $BENCH_CURRENT_NUM_RUN $query $NATIVE_INPUT_DIR $NATIVE_OUTPUT_DIR $NATIVE_FORMAT $NATIVE_OUT_FORMAT $NATIVE_SPRINT $NATIVE_FILTER" "time"
+  elif [[ "$NATIVE_FORMAT" == "tbl" ]]; then
+  execute_spark "tpch_query_$query" "--class main.scala.TpchQuery $NATIVE_SPARK_LOCAL_DIR/spark-tpc-h-queries_2.11-1.0-tbl.jar $BENCH_CURRENT_NUM_RUN $query $NATIVE_OUTPUT_DIR $NATIVE_OUT_FORMAT $NATIVE_SPRINT $NATIVE_FILTER $NATIVE_DB" "time"
+  else
```

Comment:
Can we put the common parts of these lines into an array to decrease the repetition?
```
local cmd=( --class main.scala.TpchQuery "$NATIVE_SPARK_LOCAL_DIR/spark-tpc-h-queries_2.11-1.0-txt.jar" )
if [[ $NATIVE_FORMAT = text ]]; then
    cmd+=( specific args for text format )
elif [[ $NATIVE_FORMAT = orc ]]; then
    # etc.
fi

execute_spark "tpch_query_$query" "$(printf '%q ' "${cmd[@]}")"
```
The command in the `tbl` branch looks like it's missing some arguments that the other ones have - is that intentional?